src/Parser/Warlock/Affliction/Modules/Features/Checklist.js

```diff
@@ -0,0 +1,198 @@
+            // TODO: maybe show time spent on max stacks
+            name: 'Wasted shards',
+            check: () => this.soulShardDetails.suggestionThresholds,
+          }),
+        ];
+      },
+    }),
+    new Rule({
+      name: 'Use your cooldowns',
+      description: 'Be mindful of your cooldowns if you are specced into them and use them when it\'s appropriate. It\'s okay to hold a cooldown for a little bit when the encounter requires it (burn phases), but generally speaking you should use them as much as you can.',
+      requirements: () => {
+        const combatant = this.combatants.selected;
+        return [
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.SOUL_HARVEST_TALENT,
+            when: combatant.hasTalent(SPELLS.SOUL_HARVEST_TALENT.id) || combatant.hasChest(ITEMS.THE_MASTER_HARVESTER.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.PHANTOM_SINGULARITY_TALENT,
+            when: combatant.hasTalent(SPELLS.PHANTOM_SINGULARITY_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.SUMMON_DOOMGUARD_UNTALENTED,
+            when: !combatant.hasTalent(SPELLS.GRIMOIRE_OF_SUPREMACY_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.SUMMON_INFERNAL_UNTALENTED,
+            when: !combatant.hasTalent(SPELLS.GRIMOIRE_OF_SUPREMACY_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.GRIMOIRE_FELHUNTER,
+            when: combatant.hasTalent(SPELLS.GRIMOIRE_OF_SERVICE_TALENT.id),
+          }),
+        ];
+      },
+    }),
+    // TODO: maybe add a rule about buffing UA with Haunt/MG
+    new Rule({
+      name: 'Use your utility and defensive spells',
+      description: <Wrapper>Use other spells in your toolkit to your advantage. For example, you can try to minimize necessary movement by using <SpellLink id={SPELLS.DEMONIC_GATEWAY_CAST.id} icon/>, <SpellLink id={SPELLS.DEMONIC_CIRCLE_TALENT.id} icon/>, <SpellLink id={SPELLS.BURNING_RUSH_TALENT.id} icon/> or mitigate incoming damage with <SpellLink id={SPELLS.UNENDING_RESOLVE.id} icon/>/<SpellLink id={SPELLS.DARK_PACT_TALENT.id} icon/>.</Wrapper>,
+      requirements: () => {
+        const combatant = this.combatants.selected;
+        return [
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.DEMONIC_CIRCLE_TALENT_TELEPORT,
+            when: combatant.hasTalent(SPELLS.DEMONIC_CIRCLE_TALENT.id),
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.UNENDING_RESOLVE,
+          }),
+          new GenericCastEfficiencyRequirement({
+            spell: SPELLS.DARK_PACT_TALENT,
+            when: combatant.hasTalent(SPELLS.DARK_PACT_TALENT.id),
+          }),
+        ];
+      },
+    }),
+    new Rule({
+      name: 'Try to avoid being inactive for a large portion of the fight',
```

Comment:
This name makes sense for healers, but I think for a DPS you can actually tell someone to be active. You could name it 'Always be casting' instead